src/main/java/org/betonquest/betonquest/compatibility/holographicdisplays/HologramLoop.java

```diff
@@ -100,73 +108,126 @@ public HologramLoop() {
                             LOG.warn(pack, "Could not parse item in " + key + " hologram: " + e.getMessage(), e);
                         } catch (final ObjectNotFoundException e) {
                             LOG.warn(pack, "Could not find item in " + key + " hologram: " + e.getMessage(), e);
                         }
-                    } else {
-                        hologram.appendTextLine(line.replace('&', '§'));
-                    }
-                }
-                if (checkInterval == 0) {
-                    holograms.put(hologram, conditions);
-                } else {
-                    final ConditionID[] conditionsList = conditions;
-                    final BukkitRunnable runnable = new BukkitRunnable() {
-                        @Override
-                        public void run() {
-                            for (final Player player : Bukkit.getOnlinePlayers()) {
-                                final String playerID = PlayerConverter.getID(player);
-                                if (!BetonQuest.conditions(playerID, conditionsList)) {
-                                    hologram.getVisibilityManager().hideTo(player);
-                                    continue;
-                                }
-                                hologram.getVisibilityManager().showTo(player);
-                            }
-                        }
-                    };
-                    runnable.runTaskTimerAsynchronously(BetonQuest.getInstance(), 20, checkInterval);
-                    runnables.put(hologram, runnable);
-                }
-            }
-        }
-        // loop the holograms to show/hide them
-        runnable = new BukkitRunnable() {
-            @Override
-            public void run() {
-                for (final Player player : Bukkit.getOnlinePlayers()) {
-                    final String playerID = PlayerConverter.getID(player);
-                    for (final Entry<Hologram, ConditionID[]> entry : holograms.entrySet()) {
-                        if (!BetonQuest.conditions(playerID, entry.getValue())) {
-                            entry.getKey().getVisibilityManager().hideTo(player);
+                    } else if (line.startsWith("top:")) {
+                        staticText = false;
+                        final Matcher validator = TOP_LINE_VALIDATOR.matcher(line);
+                        if (!validator.matches()) {
+                            LOG.warn("Malformed top hologram line! Expected format: 'top:<point>;<order>;<limit>[;<color>][;<color>][;<color>][;<color>]'.");
                             continue;
                         }
-                        entry.getKey().getVisibilityManager().showTo(player);
+
+                        String pointName = validator.group(1);
+                        if (!pointName.contains(".")) {
+                            pointName = new StringBuilder(packName).append('.').append(pointName).toString();
```

Comment:
A online String builder is not more performant, than simply chaining the strings